pytest-server-fixtures/pytest_server_fixtures/s3.py

```diff
@@ -0,0 +1,94 @@
+# coding: utf-8
+"""
+Pytest fixtures to launch a minio S3 server and get a bucket for it.
+"""
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import uuid
+
+import boto3
+import botocore.client
+import pytest
+from future.utils import text_type
+from pytest_fixture_config import requires_config
+
+from . import CONFIG
+from .base import TestServer
+
+
+def _s3_server(request):
+    server = MinioServer()
+    server.start()
+    request.addfinalizer(server.teardown)
+    return server
+
+
+@requires_config(CONFIG, ['minio_bin'])
+@pytest.fixture(scope="session")
+def s3_server(request):
+    return _s3_server(request)
+
+
+# Minio is a little too slow to start for each function call
+# Start it once per session and get a new bucket for each function instead.
+@pytest.fixture(scope="function")
+def s3_bucket(s3_server):  # pylint: disable=redefined-outer-name
+    client = s3_server.get_s3_client()
+    bucket_name = text_type(uuid.uuid4())
+    client.create_bucket(Bucket=bucket_name)
+    return client, bucket_name
```

Comment:
I'm a little wary of returning a tuple here, I'm not sure if it's idiomatic for pytest fixtures. Can the bucket that was created be read back from the client, like from an attribute like client.bucket_name ? Would we expect a test to use more than one bucket in each test per client?

Reply:
It's possible that a user would want to use multiple buckets, in which case this fixture isn't super useful. I'd like to avoid putting it on an attribute though.

I'll change this to a namedtuple so there's a more object-style api